### PR TITLE
Add vdirsyncer to CI

### LINKED ISSUE #38: run vdirsyncer tests from travis
It would be neat to run the vdirsyncer tests against each xandikos commit, to prevent regressions. What's the easiest way of doing this?

### PR DESCRIPTION
**Pull request by untitaker**
Fix #38

### CODE REVIEW

**Comment by jelmer** [comment]
Thanks!

Sent you a PR to fix py.test-3 running on my development machine. https://github.com/pimutils/vdirsyncer/pull/608

File: compat/xandikos-vdirsyncer.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+REPO_DIR="$(readlink -f $(dirname $0)/..)"
+TMPDIR="$(mktemp -d)"
+
+cd "$TMPDIR"
+git clone https://github.com/pimutils/vdirsyncer
+cd vdirsyncer
+make \
+    COVERAGE=true \
+    PYTEST_ARGS="--cov-config $REPO_DIR/.coveragerc --cov-append --cov xandikos tests/storage/dav/" \
```
Comment (jelmer) [inline compat/xandikos-vdirsyncer.sh]: The coverage support doesn't seem to work if the codecov bot is to be believed.

**Comment by untitaker** [comment]
Why do you want that code to do virtualenv management? Usually you just have a
virtualenv per project that you automatically enable, for Travis it shouldn't
matter.

**Comment by jelmer** [comment]
I don't use virtualenv for Xandikos development (and generally avoid it). But I'd still like to run the same tests that run in travis, e.g. before pushing.

**Comment by jelmer** [comment]
Merged, thanks. I added a few extra lines to set up a virtualenv when not running in a virtualenv.

**Pull request closed without merging**